activerecord/test/cases/adapter_test.rb

```diff
@@ -651,7 +651,65 @@ def teardown
         assert_operator Post.count, :>, 0
       end
 
+      test "can reconnect and retry queries under timeout limit when retries are enabled" do
```

Comment:
Instead of skipping, we usually wrap the tests in a conditional using `unless current_adapter?(:SQLite3Adapter)`. Also are we skipping these because the sqlite3 memory tests were failing? If it was just those that need to be skipped you can use `unless in_memory_db?` to conditionally not run these tests for that client.

Reply:
I was skipping sqlite3 because I couldn't find a good way to set up the test, given that sqlite3 doesn't have built-in SLEEP / kill mechanisms.. If we're opting for a normal sleep and testing `with_raw_connection` directly I think we should be able to test SQLite3, so let me revisit 😄